### PR TITLE
task separated

### PR DESCRIPTION
**Pull request by TexHaPb85**

### CODE REVIEW

File: src/main/java/Demonstrator.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import fire.department.Call;
+import fire.department.FireDepartment;
+
+public class Demonstrator {
+    public static void demostrateWorkOfFireDepartment() {
```
Comment (omilovskiy) [inline src/main/java/Demonstrator.java:6]: add empty line above

File: src/main/java/Demonstrator.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        FireDepartment fd = new FireDepartment();
```
Comment (omilovskiy) [inline src/main/java/Demonstrator.java]: `fireDepartment `

File: src/main/java/fire/department/FireDepartment.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    private List<Fireman> firemenStuff;
+    private Queue<Call> callQueue;
+
+    public FireDepartment() {
+        fillFiremenStuff();
+        callQueue = new LinkedList<>();
+    }
+
+    private void fillFiremenStuff() {
+        this.firemenStuff = new ArrayList<>();
```
Comment (omilovskiy) [inline src/main/java/fire/department/FireDepartment.java]: move this into constructor

File: src/main/java/fire/department/FireDepartment.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
+package fire.department;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+
+public class FireDepartment {
+    private final int NUMBER_OF_FIREMEN = 10;
+    private List<Fireman> firemenStuff;
```
Comment (omilovskiy) [inline src/main/java/fire/department/FireDepartment.java]: `firefighters`

File: src/main/java/fire/department/Call.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package fire.department;
+
+public class Call {
```
Comment (omilovskiy) [inline src/main/java/fire/department/Call.java:5]: why are you creating a class with one field inside?

File: src/main/java/fire/department/FireDepartment.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+    private void fillFiremenStuff() {
+        this.firemenStuff = new ArrayList<>();
+        for (int i = 0; i < NUMBER_OF_FIREMEN; i++) {
+            firemenStuff.add(new Fireman("Fireman" + i));
+        }
+    }
+
+    public boolean respondNewCall(Call call) {
+        for (Fireman fireman : firemenStuff) {
```
Comment (omilovskiy) [inline src/main/java/fire/department/FireDepartment.java]: use stream

File: src/main/java/fire/department/FireDepartment.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    private void fillFiremenStuff() {
+        this.firemenStuff = new ArrayList<>();
+        for (int i = 0; i < NUMBER_OF_FIREMEN; i++) {
+            firemenStuff.add(new Fireman("Fireman" + i));
+        }
+    }
+
+    public boolean respondNewCall(Call call) {
```
Comment (omilovskiy) [inline src/main/java/fire/department/FireDepartment.java]: why the return type is boolean?

File: src/main/java/fire/department/FireDepartment.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+        checkCallQueue();
+        return false;
+    }
+
+    private boolean containsFreeFireman() {
+        return firemenStuff.stream().anyMatch(Fireman::isFree);
+    }
+
+    private void checkCallQueue() {
+        while (!callQueue.isEmpty()) {
```
Comment (omilovskiy) [inline src/main/java/fire/department/FireDepartment.java:48]: add empty line above

File: src/main/java/fire/department/Fireman.java [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    private Call call;
+
+    public Fireman(String name) {
+        this.name = name;
+        this.isFree = true;
+        call = null;
+    }
+
+    private void goToTheCall() {
+        System.out.println("new call for " + name);
```
Comment (omilovskiy) [inline src/main/java/fire/department/Fireman.java:17]: as previous

File: src/main/java/fire/department/Fireman.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+        for (int i = 0; i < 5; i++) {
```
Comment (omilovskiy) [inline src/main/java/fire/department/Fireman.java:18]: as previous

File: src/main/java/fire/department/Fireman.java [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+            System.out.println("FireMan " + name
+                    + ", is going to call" + call.toString()
+                    + " from threat " + Thread.currentThread());
+            try {
```
Comment (omilovskiy) [inline src/main/java/fire/department/Fireman.java:24]: as previous

File: src/main/java/fire/department/Fireman.java [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+        call = null;
```
Comment (omilovskiy) [inline src/main/java/fire/department/Fireman.java:31]: as previous

File: src/main/java/fire/department/Fireman.java [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+        isFree = true;
+        System.out.println("FireMan " + name + " is free now");
```
Comment (omilovskiy) [inline src/main/java/fire/department/Fireman.java:33]: as previous

File: src/main/java/Demonstrator.java (hunk 1 above)
Comment (TexHaPb85) [reply src/main/java/Demonstrator.java:6]: added

File: src/main/java/Demonstrator.java (hunk 2 above)
Comment (TexHaPb85) [reply src/main/java/Demonstrator.java]: renamed

File: src/main/java/fire/department/FireDepartment.java (hunk 3 above)
Comment (TexHaPb85) [reply src/main/java/fire/department/FireDepartment.java]: moved

File: src/main/java/fire/department/FireDepartment.java (hunk 4 above)
Comment (TexHaPb85) [reply src/main/java/fire/department/FireDepartment.java]: renamed

File: src/main/java/fire/department/Call.java (hunk 5 above)
Comment (TexHaPb85) [reply src/main/java/fire/department/Call.java:5]: shell I replace it by String?

File: src/main/java/fire/department/FireDepartment.java (hunk 6 above)
Comment (TexHaPb85) [reply src/main/java/fire/department/FireDepartment.java]: done

File: src/main/java/fire/department/FireDepartment.java (hunk 8 above)
Comment (TexHaPb85) [reply src/main/java/fire/department/FireDepartment.java:48]: done

File: src/main/java/fire/department/FireDepartment.java (hunk 7 above)
Comment (TexHaPb85) [reply src/main/java/fire/department/FireDepartment.java]: fixed

File: src/main/java/fire/department/Fireman.java (hunk 9 above)
Comment (TexHaPb85) [reply src/main/java/fire/department/Fireman.java:17]: done

File: src/main/java/fire/department/Fireman.java (hunk 10 above)
Comment (TexHaPb85) [reply src/main/java/fire/department/Fireman.java:18]: done

File: src/main/java/fire/department/Fireman.java (hunk 11 above)
Comment (TexHaPb85) [reply src/main/java/fire/department/Fireman.java:24]: done

File: src/main/java/fire/department/Fireman.java (hunk 12 above)
Comment (TexHaPb85) [reply src/main/java/fire/department/Fireman.java:31]: done

File: src/main/java/fire/department/Fireman.java (hunk 13 above)
Comment (TexHaPb85) [reply src/main/java/fire/department/Fireman.java:33]: +

**Review by omilovskiy — APPROVED** [review]

**Pull request merged**